### PR TITLE
Set up a 'Hello world' project as one app with Webpack

### PR DESCRIPTION
**Pull request by abdullah-asghar-khan**
**I completed the following:**

- Create a new Rails app called 'hello-rails-react'.

- Use the commend rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.

- Verify that the jsbundling-rails gem is added to your Gemfile.

- Initialize your project with Git.

- Install needed node dependencies as described in the previous lesson's tutorial.

- Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.

- Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.

- Make sure that your project has a Postgres database set up.

- Create a table for storing your messages and create 5 different greetings.

- Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).

- Create a static view that will be the root of your app.

- Create the App component with react-router and render it in your static view.

- Create the Greeting component that will display a greeting. Set it up as a route in your App component.

- Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.

- Display the random greeting in your Greeting component.

### CODE REVIEW

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @Abdullah2213565 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Git flow was correctly used 
✔️  Good PR title and Description



## Required Changes ♻️

- [ ] Kindly make sure that your project has a Postgres database set up, this is part of the requirements, your configuration file  in[ database,yml ](https://github.com/Abdullah2213565/rails-react-app2/blob/fa7e8e981676790a2f1f62e419bb7dcd24fa4b64/config/database.yml#L7) is using sqlite, please update to Postgres 👍🏿 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @Abdullah2213565     :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Great job addressing some of the changes that were requested by a previous reviewer :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

- [ ] After following through all the steps you provided in your README to setup the project, everything seemed to go on perfectly fine until I run rails s to view the application locally and got the following error. Kindly look into why this is happening and resolve it

image


_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Abdullah2213565** [comment]
What command did you use to run the project? @Nemwel-Boniface

**Comment by prolajumokeoni** [comment]
Hi @Abdullah2213565, Kindly update your readme key and future features content 👍🏿

**Comment by abdullah-asghar-khan** [comment]
Hi @prolajumokeoni , I have done the changes required

**Comment by prolajumokeoni** [comment]
Hi @Abdullah2213565 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have a descriptive and professional readme, all sections from the microverse readme template should be present in your readme 👍🏿 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/seeds.rb [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
+
 # This file should contain all the record creation needed to seed the database with its default values.
 # The data can then be loaded with the bin/rails db:seed command (or created alongside the database with db:setup).
 #
 # Examples:
 #
 #   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
 #   Character.create(name: "Luke", movie: movies.first)
+
+Greeting.create(message: 'Hello, world!')
```
Comment (Davidosky007) [inline db/seeds.rb:11]: * As mentioned by the previous CR i still get errors trying to run the app following instructions on your readme file 
* But i was able to figure out the issue 
* The config for webpack and babel is not correct [Kindly check the lesson link and add the correct config](https://betterprogramming.pub/react-with-rails-2022-bd28e1fcd355#ad0a)
* Bellow are screenshots of my results as it finally worked and the requirement for babel config 👇 
image
image
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Abdullah2213565 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* No linter errors 👍 
* Well written PR 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AnselemOdims — APPROVED** [review]
### Hi  @Abdullah2213565 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**